tests/unit/api/objectPut.js

```diff
@@ -278,6 +278,34 @@ describe('objectPut API', () => {
                 });
         });
     });
+
+    it('should return BadDigest error and not leave orphans in data when ' +
+    'contentMD5 and completedHash do not match', done => {
+        const testPutObjectRequest = new DummyRequest({
+            bucketName,
+            namespace,
+            objectKey: objectName,
+            headers: {},
+            url: `/${bucketName}/${objectName}`,
+            contentMD5: 'vnR+tLdVF79rPPfF+7YvOg==',
+        }, Buffer.from('I am another body', 'utf8'));
+
+        bucketPut(authInfo, testPutBucketRequest, log, () => {
+            objectPut(authInfo, testPutObjectRequest, undefined, log,
+            err => {
+                assert.deepStrictEqual(err, errors.BadDigest);
+                // orphan objects don't get deleted
+                // until the next tick
+                // in memory
+                process.nextTick(() => {
+                    // Data store starts at index 1
```

Comment:
can you please run the cleanup() function from helpers.js before the test to make sure you are checking the correct indexes?  the test before this one seems to be checking ds[2] so if you are putting more data, it might be in ds[3]?  basically, you should run cleanup() before the test and make sure all of the ds is empty.